File: java/study/src/main/java/com/cs/arrays/ContainsDuplicate.java
package com.cs.arrays;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;
import java.util.concurrent.atomic.AtomicBoolean;

/**
 * Given an integer array nums, return true if any value appears at least twice in the array, a
 *  nd return false if every element is distinct.
 */
public class ContainsDuplicate {
    public static boolean containsDuplicate(int[] nums) {
        Set<Integer> memo = new HashSet<>(nums.length);
        for (int num: nums) {
            if (!memo.add(num)) {
                return true;
            }
        }

        return false;
    }

    public static boolean duplicate(int[] nums) {
        var memo = new HashSet<Integer>(nums.length);
        var ans = Arrays.stream(nums)
            .mapToObj(memo::add)
            .filter(x -> x)
            .findFirst();
        return ans.orElse(false);
    }
}

File: python/neetcode/arrays/contains_duplicate.py

def contains_duplicate(nums: list[int]) -> bool:
    """
    Given an integer array nums, return true if any value appears at least twice in the array, 
    and return false if every element is distinct.
    """

    memo = set()
    for n in nums:
        if n in memo:
            return True
        memo.add(n)
    return False

if __name__ == "__main__":
    assert contains_duplicate([1,2,3,1])
    assert contains_duplicate([1,2,3,4]) is False
    assert contains_duplicate([1,1,1,3,3,5,3,2,4,2])
